### ISSUE TITLE
ColumnDef option 'filterCellFiltered' not filtering as expected

### ISSUE THREAD

**Issue by anilnatha** [angular-ui/ui-grid#3981]
I have encountered what I believe to be a bug, but am also looking for your input in the event that I haven't setup my grid correctly.

In my grid, I have the following columndef setup for one of my columns:

	{
		'cellFilter': 'nameFilter',
		'displayName': 'Point of Contact',
		'filterCellFiltered': true,
		'name': 'pointOfContact',
		'type': 'string',
		'width': 303
	},

The data in `pointOfContact` is an Object that looks like so:

	"pointOfContact":{
		"firstName":"JOHN",
		"middleName":"",
		"lastName":"DOE"
	}

The `nameFilter`, which works as expected, is:

	var nameFilter = function( $filter ) {
		return function( input ) {
			return input.firstName + ' ' + input.lastName;
		};
	};

Yet the [`filterCellFiltered`](http://ui-grid.info/docs/#/api/ui.grid.class:GridOptions.columnDef#filtercellfiltered) option is not working as expected. I am expecting it to allow me to enter text into the filter field where the filter is applied to the _rendered_ text, not the cell's raw data, which is an object. Is my expectation of how this feature should work incorrect or should it be filtering based on the rendered contents _after_ the `cellFilter` is applied? At the present time, when I enter text into the filter, no records are found. I read through the [filtering tutorial](http://ui-grid.info/docs/#/tutorial/103_filtering) which shows an example of the `filterCellFiltered` option in use, but didn't see anything in there that I hadn't implemented correctly. On this tutorial it describes the following:

>By default the filtering will not use the formatted value after applying cellFilters, it uses the raw value from the row. The filterCellFiltered columnDef option will cause filtering to be applied after the cellFilters are evaluated, as seen on the "Long Date" field.

My development machine is running the following:

Windows 7 Enterprise
UI-Grid: v3.0.0-rc.22
Angular: 1.3.16
Browsers:
 1. Chrome (Version 43.0.2357.130 m)
 2. Firefox ESR (version 31.7) which is equivalent to Firefox 38

**Comment by anilnatha** [comment]
After doing some more investigation I think I may have found the reason behind the issue I am encountering.

I performed a diff of the version of UI-Grid I am using, [v3.0.0-rc.22](https://github.com/angular-ui/ui-grid.info/releases/tag/v3.0.0-rc.22) against the [latest unstable](http://ui-grid.info/release/ui-grid-unstable.js) version. I found that in the version I am using, there is _no_ code related to the `filterCellFiltered` property; however, in the latest unstable version there is. So it may very well be that the functionality for the `filterCellFiltered` wasn't yet added to v3.0.0-rc.22 release that I am using.
